### PR TITLE
Miner updates.

### PR DESCRIPTION
**Pull request by TimTinkers**
Had to do it on this branch due to login changes.
MiniZ ditched in favor of GMiner for now due to logging issues.
Multiple NiceHash accounts included.

### CODE REVIEW

File: packages/web-app/src/modules/salad-bowl/PluginDefinitionFactory.ts [hunk 1]
```diff
@@ -23,54 +25,56 @@ export const getPluginDefinitions = (store: RootStore): PluginDefinition[] => {
...
+    pluginDefinitions.push(getGminerCuckARoom29Definition(nicehashMiningAddress, machine)) // NiceHash
   }
 
   // BitCoinGold / ZHash
   if (has2gbSupport) {
-    pluginDefinitions.push(getGminerZHashDefinition(machine)) // NiceHash
+    pluginDefinitions.push(getGminerZHashDefinition(nicehashMiningAddress, machine)) // NiceHash
   }
 
   // Beam // BeamHashII
```
Comment (seniorquico) [inline packages/web-app/src/modules/salad-bowl/PluginDefinitionFactory.ts:60]:
```suggestion
  // Beam // BeamHashIII
```

File: packages/desktop-app/src/salad-bowl/Plugin.ts [hunk 2]
```diff
@@ -20,6 +20,10 @@ export class Plugin {
   /** The path to the location of the installed plugin */
   public readonly pluginDirectory: string
 
+  get version(): string {
+    return this.pluginDefinition.version
+  }
+
```
Comment (seniorquico) [inline packages/desktop-app/src/salad-bowl/Plugin.ts:26]: These changes are a mistake merging with the upstream branch and must be removed.

File: packages/web-app/src/modules/salad-bowl/definitions/getMiniZBeamBitflyDefinition.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { PluginDefinition } from '../models'
+import { BEAM_WALLET_ADDRESS, STANDARD_ERRORS } from './constants'
+
+export const getMiniZBeamBitflyDefinition = (machine: Machine): PluginDefinition => {
+  let def = {
+    name: 'MiniZ',
+    version: '1.6v3',
+    downloadUrl: 'https://github.com/SaladTechnologies/plugin-downloads/releases/download/miniz-1.6v3/miniz-1.6v3-cuda.zip',
+    exe: 'miniZ.exe',
+    args: `--url=ssl://${BEAM_WALLET_ADDRESS}.${machine.minerId}@us1-beam.flypool.org:3443 --par=144,5s`,
```
Comment (seniorquico) [inline packages/web-app/src/modules/salad-bowl/definitions/getMiniZBeamBitflyDefinition.ts:11]: Why aren't we adding both the `us1` and `eu1` regional servers to match the existing plugin definition?

File: packages/web-app/src/modules/salad-bowl/definitions/getMiniZBeamNiceHashDefinition.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { PluginDefinition } from '../models'
+import { STANDARD_ERRORS } from './constants'
+
+export const getMiniZBeamNiceHashDefinition = (nicehashAddress: string, machine: Machine): PluginDefinition => {
+  let def = {
+    name: 'MiniZ',
+    version: '1.6v3',
+    downloadUrl: 'https://github.com/SaladTechnologies/plugin-downloads/releases/download/miniz-1.6v3/miniz-1.6v3-cuda.zip',
+    exe: 'miniZ.exe',
+    args: `--url=tcp://${nicehashAddress}.${machine.minerId}@beamv3.usa.nicehash.com:3387 --par=144,5s`,
```
Comment (seniorquico) [inline packages/web-app/src/modules/salad-bowl/definitions/getMiniZBeamNiceHashDefinition.ts:11]: Why aren't we adding both the `usa` and `eu` regional servers to match the existing plugin definition?

File: packages/web-app/src/modules/salad-bowl/definitions/getGminerZHashDefinition.ts [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-import { NICEHASH_MINING_ADDRESS, STANDARD_ERRORS } from './constants'
+import { STANDARD_ERRORS } from './constants'
 
-const zhashUser = (location: string, minerId: string) =>
-  `-s zhash.${location}.nicehash.com -n 3369 -u ${NICEHASH_MINING_ADDRESS}.${minerId}`
+const zhashUser = (address: string, location: string, minerId: string) =>
+  `-s zhash.${location}.nicehash.com -n 3369 -u ${address}.${minerId}`
 
-export const getGminerZHashDefinition = (machine: Machine): PluginDefinition => {
+export const getGminerZHashDefinition = (nicehashAddress: string, machine: Machine): PluginDefinition => {
```
Comment (seniorquico) [inline packages/web-app/src/modules/salad-bowl/definitions/getGminerZHashDefinition.ts:8]: Shouldn't we also migrate the ZHash plugin definition from Gminer to miniZ?

File: packages/web-app/src/modules/salad-bowl/PluginDefinitionFactory.ts [hunk 6]
```diff
@@ -23,54 +25,56 @@ export const getPluginDefinitions = (store: RootStore): PluginDefinition[] => {
...
-  if (preferNiceHash && has4gbSupport) {
-    pluginDefinitions.push(getGminerBeamNiceHashDefinition(machine)) // NiceHash
+  if (has3gbSupport) {
```
Comment (seniorquico) [inline packages/web-app/src/modules/salad-bowl/PluginDefinitionFactory.ts:61]: It's miniZ that drops the VRAM requirements. This shouldn't be changed yet:

```suggestion
  if (has4gbSupport) {
```

Comment (TimTinkers) [reply packages/web-app/src/modules/salad-bowl/PluginDefinitionFactory.ts:61]: Good catch.

File: packages/web-app/src/modules/salad-bowl/definitions/getGminerZHashDefinition.ts (hunk 5 above)
Comment (TimTinkers) [reply packages/web-app/src/modules/salad-bowl/definitions/getGminerZHashDefinition.ts:8]: Added another dormant MiniZ plugin for ZHash; it's also unusable until we figure out why MiniZ just isn't giving us logs to look for the start signal in.

**Comment by TimTinkers** [comment]
> Have these changes been tested?

Yes. I ran it locally and killed the miners to see appropriate fallback with chopping.

**Pull request merged**